datastore/data_quality/quality_data.py

```diff
@@ -324,21 +324,30 @@ def get_pc_quality_grants(self):
         )
 
         for metric, query in self.quality_query_parameters.items():
-            # Aggregate total number of errors for the metric
-            ret[metric] = (
-                self.source_file_set.filter(**query)
-                .annotate(total=RawSQL("((aggregate->>%s)::int)", ("count",)))
-                .aggregate(Sum("total"))["total__sum"]
-            )
+            # Aggregate total count of errors for the metrics
+
+            # We need to drop down to SQL to do the casting/aggregates of the number of grants that are
+            # affected. Extract column/key from django orm syntax
+            # e.g. for hasRecipientIndividualsCodelists:
+            # {'quality__IndividualsCodeListsNotPresent__fail': False} extract "IndividualsCodeListsNotPresent"
+            metric_dqt_name = list(query.keys())[0].split("__")[1]
+
+            ret[metric] = self.source_file_set.annotate(
```

Comment:
I was confused initially what `ret[metric]` means - here is it the total number of grants that *don't* meet the quality criteria? But then below, it is changed to mean the percent of relevent grants that *do* mean the quality criteria?
If so, perhaps use a temporary var instead for the first use e.g. `metric_total_failed_grants`? then just set the actual rat at the end of the loop

Reply:
This was how it was already however I agree should be refactored (boo hiss variables being reused for different values etc).

I've squashed this change into the commit:

```diff
diff --git a/datastore/data_quality/quality_data.py b/datastore/data_quality/quality_data.py
index 82e5229..5536f2c 100644
--- a/datastore/data_quality/quality_data.py
+++ b/datastore/data_quality/quality_data.py
@@ -332,22 +332,23 @@ class SourceFilesStats(object):
             # {'quality__IndividualsCodeListsNotPresent__fail': False} extract "IndividualsCodeListsNotPresent"
             metric_dqt_name = list(query.keys())[0].split("__")[1]
 
-            ret[metric] = self.source_file_set.annotate(
+            metric_total_errors_count = self.source_file_set.annotate(
                 total=RawSQL("((quality->%s->>'count')::numeric)", (metric_dqt_name,))
             ).aggregate(Sum("total"))["total__sum"]
 
-            if ret[metric] == None:
-                ret[metric] = 0
+            # Guard as the query can technically return None
+            if metric_total_errors_count == None:
+                metric_total_errors_count = 0
 
             # Workout percentage of total which don't have errors / all grants in the relevant recipient set
             if "Org" in metric:
                 ret[metric] = round(
-                    (total_recipient_org_grants - ret[metric])
+                    (total_recipient_org_grants - metric_total_errors_count)
                     / total_recipient_org_grants
                     * 100
                 )
             else:
-                ret[metric] = round((total_grants - ret[metric]) / total_grants * 100)
+                ret[metric] = round((total_grants - metric_total_errors_count) / total_grants * 100)
 
         return ret
```